{
    "Modified": "2013-05-14T23:08:53.080000",
    "Published": "2010-05-27T15:30:01.767000",
    "access": {
        "authentication": "NONE",
        "complexity": "LOW",
        "vector": "NETWORK"
    },
    "capec": [],
    "cvss": 5.0,
    "cvss-time": "2010-05-28T10:47:00",
    "cwe": "CWE-189",
    "id": "CVE-2010-1634",
    "impact": {
        "availability": "PARTIAL",
        "confidentiality": "NONE",
        "integrity": "NONE"
    },
    "nessus": [
        {
            "NASL family": "Misc.",
            "NASL id": "VMWARE_ESXI_5_0_BUILD_608089_REMOTE.NASL",
            "description": "The remote VMware ESXi 5.0 host is affected by multiple vulnerabilities :\n\n  - A denial of service vulnerability exists in the     big2_toUtf8() function in file lib/xmltok.c in the     libexpat library. A remote attacker can exploit this,     via an XML document having malformed UTF-8 sequences, to     cause a buffer over-read, thus crashing the application.\n    (CVE-2009-3560)\n\n  - A denial of service vulnerability exists in the     updatePosition() function in file lib/xmltok.c in the     libexpat library. A remote attacker can exploit this,     via an XML document having malformed UTF-8 sequences, to     cause a buffer over-read, thus crashing the application.\n    (CVE-2009-3720)\n\n  - An integer overflow condition exists in the     BZ2_decompress() function in file decompress.c in the     bzip2 and libbzip2 library. A remote attacker can     exploit this, via a crafted compressed file, to cause     a denial of service or the execution of arbitrary code.\n    (CVE-2010-0405)\n\n  - A denial of service vulnerability exists in the audioop     module due to multiple integer overflows conditions in     file audioop.c. A remote attacker can exploit this, via     a large fragment or argument, to cause a buffer     overflow, resulting in an application crash.\n    (CVE-2010-1634)\n\n  - A denial of service vulnerability exists in the audioop     module due to a failure to verify the relationships     between size arguments and byte string length. A remote     attacker can exploit this, via crafted arguments, to     cause memory corruption, resulting in an application     crash. (CVE-2010-2089)\n\n  - A flaw exists in the urllib and urllib2 modules due to     processing Location headers that specify redirection to     a file. A remote attacker can exploit this, via a     crafted URL, to gain sensitive information or cause a     denial of service. (CVE-2011-1521)\n\n  - A privilege escalation vulnerability exists due to an     incorrect ACL being used for the VMware Tools folder. An     attacker on an adjacent network with access to a guest     operating system can exploit this to gain elevated     privileges on the guest operating system.\n    (CVE-2012-1518)",
            "last seen": "2019-02-21",
            "modified": "2018-11-15",
            "plugin id": "70881",
            "published": "2013-11-13",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=70881",
            "title": "ESXi 5.0 < Build 608089 Multiple Vulnerabilities (remote check)"
        },
        {
            "NASL family": "MacOS X Local Security Checks",
            "NASL id": "MACOSX_10_7_2.NASL",
            "description": "The remote host is running a version of Mac OS X 10.7.x that is prior to 10.7.2. This version contains numerous security-related fixes for the following components :\n\n  - Apache\n  - Application Firewall\n  - ATS\n  - BIND\n  - Certificate Trust Policy\n  - CFNetwork\n  - CoreMedia\n  - CoreProcesses\n  - CoreStorage\n  - File Systems\n  - iChat Server\n  - Kernel\n  - libsecurity\n  - Open Directory\n  - PHP\n  - python\n  - QuickTime\n  - SMB File Server\n  - X11",
            "last seen": "2019-02-21",
            "modified": "2018-07-14",
            "plugin id": "56480",
            "published": "2011-10-13",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=56480",
            "title": "Mac OS X 10.7.x < 10.7.2 Multiple Vulnerabilities"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-1596-1.NASL",
            "description": "It was discovered that Python would prepend an empty string to sys.path under certain circumstances. A local attacker with write access to the current working directory could exploit this to execute arbitrary code. (CVE-2008-5983)\n\nIt was discovered that the audioop module did not correctly perform input validation. If a user or automated system were tricked into opening a crafted audio file, an attacker could cause a denial of service via application crash. (CVE-2010-1634, CVE-2010-2089)\n\nGiampaolo Rodola discovered several race conditions in the smtpd module. A remote attacker could exploit this to cause a denial of service via daemon outage. (CVE-2010-3493)\n\nIt was discovered that the CGIHTTPServer module did not properly perform input validation on certain HTTP GET requests. A remote attacker could potentially obtain access to CGI script source files.\n(CVE-2011-1015)\n\nNiels Heinen discovered that the urllib and urllib2 modules would process Location headers that specify a redirection to file: URLs. A remote attacker could exploit this to obtain sensitive information or cause a denial of service. This issue only affected Ubuntu 11.04.\n(CVE-2011-1521)\n\nIt was discovered that SimpleHTTPServer did not use a charset parameter in the Content-Type HTTP header. An attacker could potentially exploit this to conduct cross-site scripting (XSS) attacks against Internet Explorer 7 users. This issue only affected Ubuntu 11.04. (CVE-2011-4940)\n\nIt was discovered that Python distutils contained a race condition when creating the ~/.pypirc file. A local attacker could exploit this to obtain sensitive information. (CVE-2011-4944)\n\nIt was discovered that SimpleXMLRPCServer did not properly validate its input when handling HTTP POST requests. A remote attacker could exploit this to cause a denial of service via excessive CPU utilization. (CVE-2012-0845)\n\nIt was discovered that Python was susceptible to hash algorithm attacks. An attacker could cause a denial of service under certian circumstances. This update adds the '-R' command line option and honors setting the PYTHONHASHSEED environment variable to 'random' to salt str and datetime objects with an unpredictable value.\n(CVE-2012-1150).\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Ubuntu security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-12-01",
            "plugin id": "62436",
            "published": "2012-10-05",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=62436",
            "title": "Ubuntu 10.04 LTS / 11.04 / 11.10 : python2.6 vulnerabilities (USN-1596-1)"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-1613-1.NASL",
            "description": "It was discovered that Python would prepend an empty string to sys.path under certain circumstances. A local attacker with write access to the current working directory could exploit this to execute arbitrary code. (CVE-2008-5983)\n\nIt was discovered that the audioop module did not correctly perform input validation. If a user or automated system were tricked into opening a crafted audio file, an attacker could cause a denial of service via application crash. (CVE-2010-1634, CVE-2010-2089)\n\nGiampaolo Rodola discovered several race conditions in the smtpd module. A remote attacker could exploit this to cause a denial of service via daemon outage. (CVE-2010-3493)\n\nIt was discovered that the CGIHTTPServer module did not properly perform input validation on certain HTTP GET requests. A remote attacker could potentially obtain access to CGI script source files.\n(CVE-2011-1015)\n\nNiels Heinen discovered that the urllib and urllib2 modules would process Location headers that specify a redirection to file: URLs. A remote attacker could exploit this to obtain sensitive information or cause a denial of service. (CVE-2011-1521)\n\nIt was discovered that SimpleHTTPServer did not use a charset parameter in the Content-Type HTTP header. An attacker could potentially exploit this to conduct cross-site scripting (XSS) attacks against Internet Explorer 7 users. (CVE-2011-4940)\n\nIt was discovered that Python distutils contained a race condition when creating the ~/.pypirc file. A local attacker could exploit this to obtain sensitive information. (CVE-2011-4944)\n\nIt was discovered that SimpleXMLRPCServer did not properly validate its input when handling HTTP POST requests. A remote attacker could exploit this to cause a denial of service via excessive CPU utilization. (CVE-2012-0845)\n\nIt was discovered that the Expat module in Python 2.5 computed hash values without restricting the ability to trigger hash collisions predictably. If a user or application using pyexpat were tricked into opening a crafted XML file, an attacker could cause a denial of service by consuming excessive CPU resources. (CVE-2012-0876)\n\nTim Boddy discovered that the Expat module in Python 2.5 did not properly handle memory reallocation when processing XML files. If a user or application using pyexpat were tricked into opening a crafted XML file, an attacker could cause a denial of service by consuming excessive memory resources. (CVE-2012-1148).\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Ubuntu security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-12-01",
            "plugin id": "62619",
            "published": "2012-10-18",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=62619",
            "title": "Ubuntu 8.04 LTS : python2.5 vulnerabilities (USN-1613-1)"
        },
        {
            "NASL family": "Solaris Local Security Checks",
            "NASL id": "SOLARIS11_PYTHON_20130313.NASL",
            "description": "The remote Solaris system is missing necessary patches to address security updates :\n\n  - Multiple integer overflows in audioop.c in the audioop     module in Python 2.6, 2.7, 3.1, and 3.2 allow     context-dependent attackers to cause a denial of service     (application crash) via a large fragment, as     demonstrated by a call to audioop.lin2lin with a long     string in the first argument, leading to a buffer     overflow. NOTE: this vulnerability exists because of an     incorrect fix for CVE-2008-3143.5. (CVE-2010-1634)",
            "last seen": "2019-02-21",
            "modified": "2018-11-15",
            "plugin id": "80748",
            "published": "2015-01-19",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=80748",
            "title": "Oracle Solaris Third-Party Patch Update : python (cve_2010_1634_integer_overflow)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_11_2_LIBPYTHON2_6-1_0-101028.NASL",
            "description": "- a race condition in the accept() implementation of     smtpd.py could lead to a denial of service     (CVE-2010-3493).\n\n  - integer overflows and insufficient size checks could     crash the audioop module (CVE-2010-2089, CVE-2010-1634).",
            "last seen": "2019-02-21",
            "modified": "2018-11-10",
            "plugin id": "53756",
            "published": "2011-05-05",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=53756",
            "title": "openSUSE Security Update : libpython2_6-1_0 (openSUSE-SU-2010:1049-1)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2010-9565.NASL",
            "description": "- Fri Jun 4 2010 David Malcolm <dmalcolm at redhat.com> -     2.6.2-8\n\n    - ensure that the compiler is invoked with '-fwrapv'       (rhbz#594819)\n\n    - CVE-2010-1634: fix various integer overflow checks in       the audioop module (patch 113)\n\n  - CVE-2010-2089: further checks within the audioop module     (patch 114)\n\n    - CVE-2008-5983: the new PySys_SetArgvEx entry point       from r81399 (patch 115)\n\n    - Fri Mar 12 2010 David Malcolm <dmalcolm at redhat.com>\n      - 2.6.2-7\n\n    - document all patches, and remove the commented-out       ones\n\n    - Address some of the issues identified in package       review (bug 226342) :\n\n    - update libs requirement on base package to use %{name}       for consistency's sake\n\n  - convert from backticks to $() syntax throughout\n\n    - wrap value of LD_LIBRARY_PATH in quotes\n\n    - convert '/usr/bin/find' requirement to 'findutils'\n\n    - remove trailing periods from summaries of subpackages\n\n    - fix spelling mistake in description of -test       subpackage\n\n    - convert usage of $$RPM_BUILD_ROOT to %{buildroot}       throughout, for stylistic consistency\n\n  - supply dirmode arguments to defattr directives\n\n    - replace references to /usr with %{_prefix}; replace       references to /usr/include with %{_includedir}\n\n  - fixup the build when __python_ver is set (Zach Sadecki;\n    bug 533989); use pybasever in the files section\n\n  - Mon Jan 25 2010 David Malcolm <dmalcolm at redhat.com> -     2.6.2-6\n\n    - update python-2.6.2-config.patch to remove downstream       customization of build of pyexpat and elementtree       modules\n\n  - add patch adapted from upstream (patch 3) to add support     for building against system expat; add\n    --with-system-expat to 'configure' invocation (patch 3)\n\n  - remove embedded copy of expat from source tree during     'prep'\n\n    - Mon Jan 25 2010 David Malcolm <dmalcolm at redhat.com>\n      - 2.6.2-5\n\n    - replace 'define' with 'global' throughout\n\n    - introduce macros for 3 directories, replacing expanded       references throughout: %{pylibdir}, %{dynload_dir},       %{site_packages}\n\n  - explicitly list all lib-dynload files, rather than     dynamically gathering the payload into a temporary text     file, so that we can be sure what we are shipping;\n    remove now-redundant testing for presence of certain .so     files\n\n  - remove embedded copy of libffi and zlib from source tree     before building\n\n    - Mon Jan 25 2010 David Malcolm <dmalcolm at redhat.com>\n      - 2.6.2-4\n\n    - change python-2.6.2-config.patch to remove our       downstream change to curses configuration in       Modules/Setup.dist, so that the curses modules are       built using setup.py with the downstream default       (linking against libncursesw.so, rather than       libncurses.so), rather than within the Makefile; add a       test to %install to verify the dso files that the       curses module is linked against the correct DSO (bug       539917; changes _cursesmodule.so -> _curses.so)\n\n  - Fri Jan 8 2010 David Malcolm <dmalcolm at redhat.com> -     2.6.2-3\n\n    - fix Lib/SocketServer.py to avoid trying to use       non-existent keyword args for os.waitpid (patch 52,       rhbz:552404, Adrian Reber)\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Fedora security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2015-10-20",
            "plugin id": "47600",
            "published": "2010-07-06",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=47600",
            "title": "Fedora 12 : python-2.6.2-8.fc12 (2010-9565)"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-1613-2.NASL",
            "description": "USN-1613-1 fixed vulnerabilities in Python 2.5. This update provides the corresponding updates for Python 2.4.\n\nIt was discovered that Python would prepend an empty string to sys.path under certain circumstances. A local attacker with write access to the current working directory could exploit this to execute arbitrary code. (CVE-2008-5983)\n\nIt was discovered that the audioop module did not correctly perform input validation. If a user or automated system were tricked into opening a crafted audio file, an attacker could cause a denial of service via application crash.\n(CVE-2010-1634, CVE-2010-2089)\n\nGiampaolo Rodola discovered several race conditions in the smtpd module. A remote attacker could exploit this to cause a denial of service via daemon outage. (CVE-2010-3493)\n\nIt was discovered that the CGIHTTPServer module did not properly perform input validation on certain HTTP GET requests. A remote attacker could potentially obtain access to CGI script source files. (CVE-2011-1015)\n\nNiels Heinen discovered that the urllib and urllib2 modules would process Location headers that specify a redirection to file: URLs. A remote attacker could exploit this to obtain sensitive information or cause a denial of service.\n(CVE-2011-1521)\n\nIt was discovered that SimpleHTTPServer did not use a charset parameter in the Content-Type HTTP header. An attacker could potentially exploit this to conduct cross-site scripting (XSS) attacks against Internet Explorer 7 users. (CVE-2011-4940)\n\nIt was discovered that Python distutils contained a race condition when creating the ~/.pypirc file. A local attacker could exploit this to obtain sensitive information.\n(CVE-2011-4944)\n\nIt was discovered that SimpleXMLRPCServer did not properly validate its input when handling HTTP POST requests. A remote attacker could exploit this to cause a denial of service via excessive CPU utilization. (CVE-2012-0845)\n\nIt was discovered that the Expat module in Python 2.5 computed hash values without restricting the ability to trigger hash collisions predictably. If a user or application using pyexpat were tricked into opening a crafted XML file, an attacker could cause a denial of service by consuming excessive CPU resources.\n(CVE-2012-0876)\n\nTim Boddy discovered that the Expat module in Python 2.5 did not properly handle memory reallocation when processing XML files. If a user or application using pyexpat were tricked into opening a crafted XML file, an attacker could cause a denial of service by consuming excessive memory resources.\n(CVE-2012-1148).\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Ubuntu security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-12-01",
            "plugin id": "62620",
            "published": "2012-10-18",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=62620",
            "title": "Ubuntu 8.04 LTS : python2.4 vulnerabilities (USN-1613-2)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2010-9652.NASL",
            "description": "- Fri Jun 4 2010 David Malcolm <dmalcolm at redhat.com> -     2.6.4-27\n\n    - ensure that the compiler is invoked with '-fwrapv'       (rhbz#594819)\n\n    - CVE-2010-1634: fix various integer overflow checks in       the audioop module (patch 113)\n\n  - CVE-2010-2089: further checks within the audioop module     (patch 114)\n\n    - CVE-2008-5983: the new PySys_SetArgvEx entry point       from r81399 (patch 115)\n\n    - Mon Apr 26 2010 Dennis Gilmore <dennis at ausil.us> -       2.6.4-26\n\n    - disable --with-valgrind on sparc arches\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Fedora security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-12",
            "plugin id": "47547",
            "published": "2010-07-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=47547",
            "title": "Fedora 13 : python-2.6.4-27.fc13 (2010-9652)"
        },
        {
            "NASL family": "VMware ESX Local Security Checks",
            "NASL id": "VMWARE_VMSA-2012-0001.NASL",
            "description": "a. ESX third-party update for Service Console kernel      The ESX Service Console Operating System (COS) kernel is updated to    kernel-2.6.18-274.3.1.el5 to fix multiple security issues in the    COS kernel.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the names CVE-2011-0726, CVE-2011-1078, CVE-2011-1079,    CVE-2011-1080, CVE-2011-1093, CVE-2011-1163, CVE-2011-1166,    CVE-2011-1170, CVE-2011-1171, CVE-2011-1172, CVE-2011-1494,    CVE-2011-1495, CVE-2011-1577, CVE-2011-1763, CVE-2010-4649,    CVE-2011-0695, CVE-2011-0711, CVE-2011-1044, CVE-2011-1182,    CVE-2011-1573, CVE-2011-1576, CVE-2011-1593, CVE-2011-1745,    CVE-2011-1746, CVE-2011-1776, CVE-2011-1936, CVE-2011-2022,    CVE-2011-2213, CVE-2011-2492, CVE-2011-1780, CVE-2011-2525,    CVE-2011-2689, CVE-2011-2482, CVE-2011-2491, CVE-2011-2495,    CVE-2011-2517, CVE-2011-2519, CVE-2011-2901 to these issues.\n  b. ESX third-party update for Service Console cURL RPM      The ESX Service Console (COS) curl RPM is updated to cURL-7.15.5.9    resolving a security issues.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the name CVE-2011-2192 to this issue.\n  c. ESX third-party update for Service Console nspr and nss RPMs      The ESX Service Console (COS) nspr and nss RPMs are updated to    nspr-4.8.8-1.el5_7 and nss-3.12.10-4.el5_7 respectively resolving    a security issues.\n     A Certificate Authority (CA) issued fraudulent SSL certificates and    Netscape Portable Runtime (NSPR) and Network Security Services (NSS)    contain the built-in tokens of this fraudulent Certificate    Authority. This update renders all SSL certificates signed by the    fraudulent CA as untrusted for all uses.\n  d. ESX third-party update for Service Console rpm RPMs      The ESX Service Console Operating System (COS) rpm packages are    updated to popt-1.10.2.3-22.el5_7.2, rpm-4.4.2.3-22.el5_7.2,    rpm-libs-4.4.2.3-22.el5_7.2 and rpm-python-4.4.2.3-22.el5_7.2    which fixes multiple security issues.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the names CVE-2010-2059 and CVE-2011-3378 to these    issues.\n  e. ESX third-party update for Service Console samba RPMs      The ESX Service Console Operating System (COS) samba packages are    updated to samba-client-3.0.33-3.29.el5_7.4,    samba-common-3.0.33-3.29.el5_7.4 and    libsmbclient-3.0.33-3.29.el5_7.4 which fixes multiple security    issues in the Samba client.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the names CVE-2010-0547, CVE-2010-0787, CVE-2011-1678,    CVE-2011-2522 and CVE-2011-2694 to these issues.\n     Note that ESX does not include the Samba Web Administration Tool    (SWAT) and therefore ESX COS is not affected by CVE-2011-2522 and    CVE-2011-2694.\n  f. ESX third-party update for Service Console python package      The ESX Service Console (COS) python package is updated to    2.4.3-44 which fixes multiple security issues.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the names CVE-2009-3720, CVE-2010-3493, CVE-2011-1015 and    CVE-2011-1521 to these issues.\n  g. ESXi update to third-party component python      The python third-party library is updated to python 2.5.6 which    fixes multiple security issues.\n     The Common Vulnerabilities and Exposures project (cve.mitre.org) has    assigned the names CVE-2009-3560, CVE-2009-3720, CVE-2010-1634,    CVE-2010-2089, and CVE-2011-1521 to these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-09-06",
            "plugin id": "57749",
            "published": "2012-01-31",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=57749",
            "title": "VMSA-2012-0001 : VMware ESXi and ESX updates to third-party library and ESX Service Console"
        },
        {
            "NASL family": "Mandriva Local Security Checks",
            "NASL id": "MANDRIVA_MDVSA-2010-132.NASL",
            "description": "Multiple vulnerabilities has been found and corrected in python :\n\nMultiple integer overflows in audioop.c in the audioop module in Ptthon allow context-dependent attackers to cause a denial of service (application crash) via a large fragment, as demonstrated by a call to audioop.lin2lin with a long string in the first argument, leading to a buffer overflow. NOTE: this vulnerability exists because of an incorrect fix for CVE-2008-3143.5 (CVE-2010-1634).\n\nThe audioop module in Python does not verify the relationships between size arguments and byte string lengths, which allows context-dependent attackers to cause a denial of service (memory corruption and application crash) via crafted arguments, as demonstrated by a call to audioop.reverse with a one-byte string, a different vulnerability than CVE-2010-1634 (CVE-2010-2089).\n\nPackages for 2008.0 and 2009.0 are provided as of the Extended Maintenance Program. Please visit this link to learn more:\nhttp://store.mandriva.com/product_info.php?cPath=149&amp;products_id=4 90\n\nThe updated packages have been patched to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-19",
            "plugin id": "48191",
            "published": "2010-07-30",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=48191",
            "title": "Mandriva Linux Security Advisory : python (MDVSA-2010:132)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_11_LIBPYTHON2_6-1_0-101109.NASL",
            "description": "The following issues have been fixed :\n\n  - a race condition in the accept() implementation of     smtpd.py could lead to a denial of service.\n    (CVE-2010-3493)\n\n  - integer overflows and insufficient size checks could     crash the audioop module. (CVE-2010-2089 /     CVE-2010-1634)",
            "last seen": "2019-02-21",
            "modified": "2013-10-25",
            "plugin id": "51623",
            "published": "2011-01-21",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=51623",
            "title": "SuSE 11.1 Security Update : Python (SAT Patch Number 3493)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_11_LIBPYTHON2_6-1_0-101028.NASL",
            "description": "The following issues have been fixed :\n\n  - a race condition in the accept() implementation of     smtpd.py could lead to a denial of service.\n    (CVE-2010-3493)\n\n  - integer overflows and insufficient size checks could     crash the audioop module. (CVE-2010-2089 /     CVE-2010-1634)",
            "last seen": "2019-02-21",
            "modified": "2013-10-25",
            "plugin id": "51134",
            "published": "2010-12-12",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=51134",
            "title": "SuSE 11 Security Update : Python (SAT Patch Number 3491)"
        },
        {
            "NASL family": "Scientific Linux Local Security Checks",
            "NASL id": "SL_20110113_PYTHON_ON_SL5_X.NASL",
            "description": "It was found that many applications embedding the Python interpreter did not specify a valid full path to the script or application when calling the PySys_SetArgv API function, which could result in the addition of the current working directory to the module search path (sys.path). A local attacker able to trick a victim into running such an application in an attacker-controlled directory could use this flaw to execute code with the victim's privileges. This update adds the PySys_SetArgvEx API. Developers can modify their applications to use this new API, which sets sys.argv without modifying sys.path.\n(CVE-2008-5983)\n\nMultiple flaws were found in the Python rgbimg module. If an application written in Python was using the rgbimg module and loaded a specially crafted SGI image file, it could cause the application to crash or, possibly, execute arbitrary code with the privileges of the user running the application. (CVE-2009-4134, CVE-2010-1449, CVE-2010-1450)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nThis update also fixes the following bugs :\n\n  - When starting a child process from the subprocess module     in Python 2.4, the parent process could leak file     descriptors if an error occurred. This update resolves     the issue. (BZ#609017)\n\n  - Prior to Python 2.7, programs that used 'ulimit -n' to     enable communication with large numbers of subprocesses     could still monitor only 1024 file descriptors at a     time, which caused an exception :\n\n    ValueError: filedescriptor out of range in select()\n\nThis was due to the subprocess module using the 'select' system call.\nThe module now uses the 'poll' system call, removing this limitation.\n(BZ#609020)\n\n  - Prior to Python 2.5, the tarfile module failed to unpack     tar files if the path was longer than 100 characters.\n    This update backports the tarfile module from Python 2.5     and the issue no longer occurs. (BZ#263401)\n\n  - The email module incorrectly implemented the logic for     obtaining attachment file names: the get_filename()     fallback for using the deprecated 'name' parameter of     the 'Content-Type' header erroneously used the     'Content-Disposition' header. This update backports a     fix from Python 2.6, which resolves this issue.\n    (BZ#644147)\n\n  - Prior to version 2.5, Python's optimized memory     allocator never released memory back to the system. The     memory usage of a long-running Python process would     resemble a 'high-water mark'. This update backports a     fix from Python 2.5a1, which frees unused arenas, and     adds a non-standard sys._debugmallocstats() function,     which prints diagnostic information to stderr. Finally,     when running under Valgrind, the optimized allocator is     deactivated, to allow more convenient debugging of     Python memory usage issues. (BZ#569093)\n\n  - The urllib and urllib2 modules ignored the no_proxy     variable, which could lead to programs such as 'yum'     erroneously accessing a proxy server for URLs covered by     a 'no_proxy' exclusion. This update backports fixes of     urllib and urllib2, which respect the 'no_proxy'     variable, which fixes these issues. (BZ#549372)\n\nAs well, this update adds the following enhancements :\n\n  - This update introduces a new python-libs package,     subsuming the majority of the content of the core python     package. This makes both 32-bit and 64-bit Python     libraries available on PowerPC systems. (BZ#625372)\n\n  - The python-libs.i386 package is now available for 64-bit     Itanium with the 32-bit Itanium compatibility mode.\n    (BZ#644761)",
            "last seen": "2019-02-21",
            "modified": "2018-12-31",
            "plugin id": "60935",
            "published": "2012-08-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=60935",
            "title": "Scientific Linux Security Update : python on SL5.x i386/x86_64"
        },
        {
            "NASL family": "MacOS X Local Security Checks",
            "NASL id": "MACOSX_SECUPD2011-006.NASL",
            "description": "The remote host is running a version of Mac OS X 10.6 that does not have Security Update 2011-006 applied.  This update contains numerous security-related fixes for the following components :\n\n  - Apache\n  - Application Firewall\n  - ATS\n  - BIND\n  - Certificate Trust Policy\n  - CFNetwork\n  - CoreFoundation\n  - CoreMedia\n  - File Systems\n  - IOGraphics\n  - iChat Server\n  - Mailman\n  - MediaKit\n  - PHP\n  - postfix\n  - python\n  - QuickTime\n  - Tomcat\n  - User Documentation\n  - Web Server\n  - X11",
            "last seen": "2019-02-21",
            "modified": "2018-07-14",
            "plugin id": "56481",
            "published": "2011-10-13",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=56481",
            "title": "Mac OS X Multiple Vulnerabilities (Security Update 2011-006)"
        },
        {
            "NASL family": "Oracle Linux Local Security Checks",
            "NASL id": "ORACLELINUX_ELSA-2011-0491.NASL",
            "description": "From Red Hat Security Advisory 2011:0491 :\n\nUpdated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 4.\n\nThe Red Hat Security Response Team has rated this update as having moderate security impact. Common Vulnerability Scoring System (CVSS) base scores, which give detailed severity ratings, are available for each vulnerability from the CVE links in the References section.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nA flaw was found in the Python urllib and urllib2 libraries where they would not differentiate between different target URLs when handling automatic redirects. This caused Python applications using these modules to follow any new URL that they understood, including the 'file://' URL type. This could allow a remote server to force a local Python application to read a local file instead of the remote one, possibly exposing local files that were not meant to be exposed.\n(CVE-2011-1521)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nA race condition was found in the way the Python smtpd module handled new connections. A remote user could use this flaw to cause a Python script using the smtpd module to terminate. (CVE-2010-3493)\n\nAn information disclosure flaw was found in the way the Python CGIHTTPServer module processed certain HTTP GET requests. A remote attacker could use a specially crafted request to obtain the CGI script's source code. (CVE-2011-1015)\n\nA buffer over-read flaw was found in the way the Python Expat parser handled malformed UTF-8 sequences when processing XML files. A specially crafted XML file could cause Python applications using the Python Expat parser to crash while parsing the file. (CVE-2009-3720)\n\nThis update makes Python use the system Expat library rather than its own internal copy; therefore, users must have the version of Expat shipped with RHSA-2009:1625 installed, or a later version, to resolve the CVE-2009-3720 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-07-18",
            "plugin id": "68270",
            "published": "2013-07-12",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=68270",
            "title": "Oracle Linux 4 : python (ELSA-2011-0491)"
        },
        {
            "NASL family": "Scientific Linux Local Security Checks",
            "NASL id": "SL_20110505_PYTHON_ON_SL4_X.NASL",
            "description": "A flaw was found in the Python urllib and urllib2 libraries where they would not differentiate between different target URLs when handling automatic redirects. This caused Python applications using these modules to follow any new URL that they understood, including the 'file://' URL type. This could allow a remote server to force a local Python application to read a local file instead of the remote one, possibly exposing local files that were not meant to be exposed.\n(CVE-2011-1521)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nA race condition was found in the way the Python smtpd module handled new connections. A remote user could use this flaw to cause a Python script using the smtpd module to terminate. (CVE-2010-3493)\n\nAn information disclosure flaw was found in the way the Python CGIHTTPServer module processed certain HTTP GET requests. A remote attacker could use a specially crafted request to obtain the CGI script's source code. (CVE-2011-1015)\n\nA buffer over-read flaw was found in the way the Python Expat parser handled malformed UTF-8 sequences when processing XML files. A specially crafted XML file could cause Python applications using the Python Expat parser to crash while parsing the file. (CVE-2009-3720)",
            "last seen": "2019-02-21",
            "modified": "2018-12-31",
            "plugin id": "61033",
            "published": "2012-08-01",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=61033",
            "title": "Scientific Linux Security Update : python on SL4.x, SL5.x i386/x86_64"
        },
        {
            "NASL family": "Red Hat Local Security Checks",
            "NASL id": "REDHAT-RHSA-2011-0491.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 4.\n\nThe Red Hat Security Response Team has rated this update as having moderate security impact. Common Vulnerability Scoring System (CVSS) base scores, which give detailed severity ratings, are available for each vulnerability from the CVE links in the References section.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nA flaw was found in the Python urllib and urllib2 libraries where they would not differentiate between different target URLs when handling automatic redirects. This caused Python applications using these modules to follow any new URL that they understood, including the 'file://' URL type. This could allow a remote server to force a local Python application to read a local file instead of the remote one, possibly exposing local files that were not meant to be exposed.\n(CVE-2011-1521)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nA race condition was found in the way the Python smtpd module handled new connections. A remote user could use this flaw to cause a Python script using the smtpd module to terminate. (CVE-2010-3493)\n\nAn information disclosure flaw was found in the way the Python CGIHTTPServer module processed certain HTTP GET requests. A remote attacker could use a specially crafted request to obtain the CGI script's source code. (CVE-2011-1015)\n\nA buffer over-read flaw was found in the way the Python Expat parser handled malformed UTF-8 sequences when processing XML files. A specially crafted XML file could cause Python applications using the Python Expat parser to crash while parsing the file. (CVE-2009-3720)\n\nThis update makes Python use the system Expat library rather than its own internal copy; therefore, users must have the version of Expat shipped with RHSA-2009:1625 installed, or a later version, to resolve the CVE-2009-3720 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-12-20",
            "plugin id": "53820",
            "published": "2011-05-06",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=53820",
            "title": "RHEL 4 : python (RHSA-2011:0491)"
        },
        {
            "NASL family": "CentOS Local Security Checks",
            "NASL id": "CENTOS_RHSA-2011-0491.NASL",
            "description": "Updated python packages that fix multiple security issues are now available for Red Hat Enterprise Linux 4.\n\nThe Red Hat Security Response Team has rated this update as having moderate security impact. Common Vulnerability Scoring System (CVSS) base scores, which give detailed severity ratings, are available for each vulnerability from the CVE links in the References section.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nA flaw was found in the Python urllib and urllib2 libraries where they would not differentiate between different target URLs when handling automatic redirects. This caused Python applications using these modules to follow any new URL that they understood, including the 'file://' URL type. This could allow a remote server to force a local Python application to read a local file instead of the remote one, possibly exposing local files that were not meant to be exposed.\n(CVE-2011-1521)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nA race condition was found in the way the Python smtpd module handled new connections. A remote user could use this flaw to cause a Python script using the smtpd module to terminate. (CVE-2010-3493)\n\nAn information disclosure flaw was found in the way the Python CGIHTTPServer module processed certain HTTP GET requests. A remote attacker could use a specially crafted request to obtain the CGI script's source code. (CVE-2011-1015)\n\nA buffer over-read flaw was found in the way the Python Expat parser handled malformed UTF-8 sequences when processing XML files. A specially crafted XML file could cause Python applications using the Python Expat parser to crash while parsing the file. (CVE-2009-3720)\n\nThis update makes Python use the system Expat library rather than its own internal copy; therefore, users must have the version of Expat shipped with RHSA-2009:1625 installed, or a later version, to resolve the CVE-2009-3720 issue.\n\nAll Python users should upgrade to these updated packages, which contain backported patches to correct these issues.",
            "last seen": "2019-02-21",
            "modified": "2018-11-10",
            "plugin id": "53814",
            "published": "2011-05-06",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=53814",
            "title": "CentOS 4 : python (CESA-2011:0491)"
        },
        {
            "NASL family": "Ubuntu Local Security Checks",
            "NASL id": "UBUNTU_USN-1616-1.NASL",
            "description": "It was discovered that Python would prepend an empty string to sys.path under certain circumstances. A local attacker with write access to the current working directory could exploit this to execute arbitrary code. This issue only affected Ubuntu 10.04 LTS.\n(CVE-2008-5983)\n\nIt was discovered that the audioop module did not correctly perform input validation. If a user or automated system were tricked into opening a crafted audio file, an attacker could cause a denial of service via application crash. These issues only affected Ubuntu 10.04 LTS. (CVE-2010-1634, CVE-2010-2089)\n\nIt was discovered that Python distutils contained a race condition when creating the ~/.pypirc file. A local attacker could exploit this to obtain sensitive information. (CVE-2011-4944)\n\nIt was discovered that SimpleXMLRPCServer did not properly validate its input when handling HTTP POST requests. A remote attacker could exploit this to cause a denial of service via excessive CPU utilization. (CVE-2012-0845)\n\nIt was discovered that Python was susceptible to hash algorithm attacks. An attacker could cause a denial of service under certian circumstances. This update adds the '-R' command line option and honors setting the PYTHONHASHSEED environment variable to 'random' to salt str and datetime objects with an unpredictable value.\n(CVE-2012-1150)\n\nSerhiy Storchaka discovered that the UTF16 decoder in Python did not properly reset internal variables after error handling. An attacker could exploit this to cause a denial of service via memory corruption.\n(CVE-2012-2135).\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Ubuntu security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2018-12-01",
            "plugin id": "62700",
            "published": "2012-10-25",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=62700",
            "title": "Ubuntu 10.04 LTS / 11.04 : python3.1 vulnerabilities (USN-1616-1)"
        },
        {
            "NASL family": "Red Hat Local Security Checks",
            "NASL id": "REDHAT-RHSA-2011-0027.NASL",
            "description": "Updated python packages that fix multiple security issues, several bugs, and add two enhancements are now available for Red Hat Enterprise Linux 5.\n\nThe Red Hat Security Response Team has rated this update as having low security impact. Common Vulnerability Scoring System (CVSS) base scores, which give detailed severity ratings, are available for each vulnerability from the CVE links in the References section.\n\nPython is an interpreted, interactive, object-oriented programming language.\n\nIt was found that many applications embedding the Python interpreter did not specify a valid full path to the script or application when calling the PySys_SetArgv API function, which could result in the addition of the current working directory to the module search path (sys.path). A local attacker able to trick a victim into running such an application in an attacker-controlled directory could use this flaw to execute code with the victim's privileges. This update adds the PySys_SetArgvEx API. Developers can modify their applications to use this new API, which sets sys.argv without modifying sys.path.\n(CVE-2008-5983)\n\nMultiple flaws were found in the Python rgbimg module. If an application written in Python was using the rgbimg module and loaded a specially crafted SGI image file, it could cause the application to crash or, possibly, execute arbitrary code with the privileges of the user running the application. (CVE-2009-4134, CVE-2010-1449, CVE-2010-1450)\n\nMultiple flaws were found in the Python audioop module. Supplying certain inputs could cause the audioop module to crash or, possibly, execute arbitrary code. (CVE-2010-1634, CVE-2010-2089)\n\nThis update also fixes the following bugs :\n\n* When starting a child process from the subprocess module in Python 2.4, the parent process could leak file descriptors if an error occurred. This update resolves the issue. (BZ#609017)\n\n* Prior to Python 2.7, programs that used 'ulimit -n' to enable communication with large numbers of subprocesses could still monitor only 1024 file descriptors at a time, which caused an exception :\n\nValueError: filedescriptor out of range in select()\n\nThis was due to the subprocess module using the 'select' system call.\nThe module now uses the 'poll' system call, removing this limitation.\n(BZ#609020)\n\n* Prior to Python 2.5, the tarfile module failed to unpack tar files if the path was longer than 100 characters. This update backports the tarfile module from Python 2.5 and the issue no longer occurs.\n(BZ#263401)\n\n* The email module incorrectly implemented the logic for obtaining attachment file names: the get_filename() fallback for using the deprecated 'name' parameter of the 'Content-Type' header erroneously used the 'Content-Disposition' header. This update backports a fix from Python 2.6, which resolves this issue. (BZ#644147)\n\n* Prior to version 2.5, Python's optimized memory allocator never released memory back to the system. The memory usage of a long-running Python process would resemble a 'high-water mark'. This update backports a fix from Python 2.5a1, which frees unused arenas, and adds a non-standard sys._debugmallocstats() function, which prints diagnostic information to stderr. Finally, when running under Valgrind, the optimized allocator is deactivated, to allow more convenient debugging of Python memory usage issues. (BZ#569093)\n\n* The urllib and urllib2 modules ignored the no_proxy variable, which could lead to programs such as 'yum' erroneously accessing a proxy server for URLs covered by a 'no_proxy' exclusion. This update backports fixes of urllib and urllib2, which respect the 'no_proxy' variable, which fixes these issues. (BZ#549372)\n\nAs well, this update adds the following enhancements :\n\n* This update introduces a new python-libs package, subsuming the majority of the content of the core python package. This makes both 32-bit and 64-bit Python libraries available on PowerPC systems.\n(BZ#625372)\n\n* The python-libs.i386 package is now available for 64-bit Itanium with the 32-bit Itanium compatibility mode. (BZ#644761)\n\nAll Python users are advised to upgrade to these updated packages, which contain backported patches to correct these issues and add these enhancements.",
            "last seen": "2019-02-21",
            "modified": "2018-11-26",
            "plugin id": "51524",
            "published": "2011-01-14",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=51524",
            "title": "RHEL 5 : python (RHSA-2011:0027)"
        },
        {
            "NASL family": "Fedora Local Security Checks",
            "NASL id": "FEDORA_2010-13388.NASL",
            "description": "- Backport from F14: - Fix for lone surrogates, utf8 and     certain encode error handlers. - Fix an incompatibility     between pyexpat and the system expat-2.0.1 that led to a     segfault running test_pyexpat.py (patch 110; upstream     issue 9054; rhbz#610312)\n\nNote that Tenable Network Security has extracted the preceding description block directly from the Fedora security advisory. Tenable has attempted to automatically clean and format it as much as possible without introducing additional issues.",
            "last seen": "2019-02-21",
            "modified": "2015-10-20",
            "plugin id": "49107",
            "published": "2010-09-04",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=49107",
            "title": "Fedora 13 : python3-3.1.2-7.fc13 (2010-13388)"
        },
        {
            "NASL family": "SuSE Local Security Checks",
            "NASL id": "SUSE_PYTHON-7314.NASL",
            "description": "With this update of Python :\n\n  - a race condition in the accept() implementation of     smtpd.py could lead to a denial of service.\n    (CVE-2010-3493)\n\n  - integer overflows and insufficient size checks could     crash the audioop and rgbimg modules. (CVE-2010-2089 /     CVE-2010-1634 / CVE-2009-4134 / CVE-2010-1449 /     CVE-2010-1450)",
            "last seen": "2019-02-21",
            "modified": "2012-05-17",
            "plugin id": "51642",
            "published": "2011-01-21",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=51642",
            "title": "SuSE 10 Security Update : Python (ZYPP Patch Number 7314)"
        },
        {
            "NASL family": "Misc.",
            "NASL id": "VMWARE_VMSA-2012-0001_REMOTE.NASL",
            "description": "The remote VMware ESX / ESXi host is missing a security-related patch.\nIt is, therefore, affected by multiple vulnerabilities, including remote code execution vulnerabilities, in several third-party libraries :\n\n  - COS kernel\n  - cURL\n  - python\n  - rpm",
            "last seen": "2019-02-21",
            "modified": "2018-08-16",
            "plugin id": "89105",
            "published": "2016-03-03",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=89105",
            "title": "VMware ESX / ESXi Service Console and Third-Party Libraries Multiple Vulnerabilities (VMSA-2012-0001) (remote check)"
        },
        {
            "NASL family": "Gentoo Local Security Checks",
            "NASL id": "GENTOO_GLSA-201401-04.NASL",
            "description": "The remote host is affected by the vulnerability described in GLSA-201401-04 (Python: Multiple vulnerabilities)\n\n    Multiple vulnerabilities have been discovered in Python. Please review       the CVE identifiers referenced below for details.\n  Impact :\n\n    A remote attacker could possibly cause a Denial of Service condition or       perform a man-in-the-middle attack to disclose sensitive information.\n  Workaround :\n\n    There is no known workaround at this time.",
            "last seen": "2019-02-21",
            "modified": "2018-07-12",
            "plugin id": "71811",
            "published": "2014-01-07",
            "reporter": "Tenable",
            "source": "https://www.tenable.com/plugins/index.php?view=single&id=71811",
            "title": "GLSA-201401-04 : Python: Multiple vulnerabilities"
        }
    ],
    "ranking": [
        [
            {
                "circl": 3
            }
        ]
    ],
    "redhat": {
        "advisories": [
            {
                "rhsa": {
                    "id": "RHSA-2011:0027"
                }
            }
        ],
        "rpms": [
            "python-0:2.4.3-43.el5",
            "python-devel-0:2.4.3-43.el5",
            "python-libs-0:2.4.3-43.el5",
            "python-tools-0:2.4.3-43.el5",
            "tkinter-0:2.4.3-43.el5",
            "python-0:2.3.4-14.10.el4",
            "python-devel-0:2.3.4-14.10.el4",
            "python-docs-0:2.3.4-14.10.el4",
            "python-tools-0:2.3.4-14.10.el4",
            "tkinter-0:2.3.4-14.10.el4"
        ]
    },
    "references": [
        "http://bugs.python.org/issue8674",
        "http://lists.apple.com/archives/Security-announce/2011//Oct/msg00003.html",
        "http://lists.fedoraproject.org/pipermail/package-announce/2010-June/042751.html",
        "http://lists.opensuse.org/opensuse-security-announce/2010-12/msg00006.html",
        "http://lists.opensuse.org/opensuse-security-announce/2011-01/msg00006.html",
        "http://support.apple.com/kb/HT5002",
        "http://svn.python.org/view?rev=81045&view=rev",
        "http://svn.python.org/view?rev=81079&view=rev",
        "http://www.redhat.com/support/errata/RHSA-2011-0027.html",
        "http://www.securityfocus.com/bid/40370",
        "http://www.ubuntu.com/usn/USN-1596-1",
        "http://www.ubuntu.com/usn/USN-1613-1",
        "http://www.ubuntu.com/usn/USN-1613-2",
        "http://www.ubuntu.com/usn/USN-1616-1",
        "http://www.vupen.com/english/advisories/2010/1448",
        "http://www.vupen.com/english/advisories/2011/0122",
        "http://www.vupen.com/english/advisories/2011/0212",
        "https://bugzilla.redhat.com/show_bug.cgi?id=590690"
    ],
    "refmap": {
        "apple": [
            "APPLE-SA-2011-10-12-3"
        ],
        "bid": [
            "40370"
        ],
        "confirm": [
            "http://bugs.python.org/issue8674",
            "http://support.apple.com/kb/HT5002",
            "http://svn.python.org/view?rev=81045&view=rev",
            "http://svn.python.org/view?rev=81079&view=rev",
            "https://bugzilla.redhat.com/show_bug.cgi?id=590690"
        ],
        "fedora": [
            "FEDORA-2010-9652"
        ],
        "secunia": [
            "39937",
            "40194",
            "42888",
            "43068",
            "50858",
            "51024",
            "51040",
            "51087"
        ],
        "suse": [
            "SUSE-SR:2010:024",
            "SUSE-SR:2011:002"
        ],
        "ubuntu": [
            "USN-1596-1",
            "USN-1613-1",
            "USN-1613-2",
            "USN-1616-1"
        ],
        "vupen": [
            "ADV-2010-1448",
            "ADV-2011-0122",
            "ADV-2011-0212"
        ]
    },
    "summary": "Multiple integer overflows in audioop.c in the audioop module in Python 2.6, 2.7, 3.1, and 3.2 allow context-dependent attackers to cause a denial of service (application crash) via a large fragment, as demonstrated by a call to audioop.lin2lin with a long string in the first argument, leading to a buffer overflow.  NOTE: this vulnerability exists because of an incorrect fix for CVE-2008-3143.5.",
    "vmware": [
        {
            "description": "The ESX Service Console Operating System (COS) samba packages are updated to samba-client-3.0.33-3.29.el5_7.4samba-common-3.0.33-3.29.el5_7.4 and libsmbclient-3.0.33-3.29.el5_7.4 which fixes multiple security issues in the Samba client.",
            "id": "VMSA-2012-0001",
            "last_updated": "2012-03-29T00:00:00",
            "published": "2012-01-30T00:00:00",
            "title": "ESX third party update for Service Console samba RPMs"
        }
    ],
    "vulnerable_configuration": [
        {
            "id": "cpe:2.3:a:python:python:2.6",
            "title": "Python 2.6"
        },
        {
            "id": "cpe:2.3:a:python:python:2.7",
            "title": "Python 2.7"
        },
        {
            "id": "cpe:2.3:a:python:python:3.1",
            "title": "Python 3.1"
        },
        {
            "id": "cpe:2.3:a:python:python:3.2",
            "title": "Python 3.2"
        }
    ],
    "vulnerable_configuration_cpe_2_2": [
        "cpe:/a:python:python:2.6",
        "cpe:/a:python:python:2.7",
        "cpe:/a:python:python:3.1",
        "cpe:/a:python:python:3.2"
    ]
}